### PR TITLE
True multi region support

### PR DESCRIPTION
**Pull request by jb4free**
This needs some testing cycles but now using only the now customer url e.g. https://company.platform9.net we and the region desired we can look up and determine the correct region fqdn for hosts.

#188

### CODE REVIEW

File: pf9-express [hunk 1]
```diff
@@ -618,9 +619,18 @@ proxy_url=$(grep ^proxy_url ${pf9_config} | cut -d \| -f2)
...
 if [ "${proxy_url}" != "-" ]; then extra_vars="${extra_vars} proxy_url=${proxy_url}"; fi
 
+## get region specific hostname
+## Get Keystone Token
+token=`curl -k -i -H "Content-Type: application/json" https://${du_url}/keystone/v3/auth/tokens?nocatalog \
+    -d "{ \"auth\": { \"identity\": { \"methods\": [\"password\"], \"password\": { \"user\": { \"name\": \"${du_username}\", \"domain\": {\"id\": \"default\"}, \"password\": \"${du_password}\" } } }, \"scope\": { \"project\": { \"name\": \"${du_tenant}\", \"domain\": {\"id\": \"default\"}}}}}" 2>/dev/null | grep -i ^X-Subject-Token | awk -F : '{print $2}' | sed -e 's/ //g' | sed -e 's/\r//g'`
+
+## get region info
+service_id=`curl -X GET -H "X-Auth-Token: ${token}" https://${du_url}/keystone/v3/services?type=regionInfo 2>/dev/null | grep -o '"id": "[a-z0-9 ]*' | cut -c 8-40`
+region_url=`curl -X GET -H "X-Auth-Token: ${token}" https://${du_url}/keystone/v3/endpoints?service_id=${service_id} | jq --arg du_region "${du_region}" '.endpoints[] | select(.region_id==$du_region and .interface=="public")' | jq '.url' | awk '{print substr($0,2,length($0)-9)}' | cut -d '/' -f3`
```
Comment (busterswt) [inline pf9-express]: jq is being used here before pf9-express has had a chance to install it:

root@platform9-jump-server:/opt/pf9-express# ./pf9-express -a pmo
./pf9-express: line 629: jq: command not found

So maybe prereqs should be run first (./pf9-express -i)

Comment (jb4free) [reply pf9-express]: were you using the -b command?

File: pf9-express [hunk 2]
```diff
@@ -618,9 +619,18 @@ proxy_url=$(grep ^proxy_url ${pf9_config} | cut -d \| -f2)
...
+
 ## assign/validate ctrl_ip from config file and resolve IP for ctrl_hostname
-ctrl_hostname=$(grep ^du_url ${pf9_config} | cut -d \| -f2 | cut -d \/ -f3)
-tmp_ip=$(ping -c 1 ${ctrl_hostname} | grep PING | cut -d ' ' -f3)
+ctrl_hostname=$( ${du_url} | cut -d \| -f2 | cut -d \/ -f3)
```
Comment (busterswt) [inline pf9-express]: This:
ctrl_hostname=$( ${du_url} | cut -d \| -f2 | cut -d \/ -f3)

Should read:
ctrl_hostname=$(echo ${du_url} | cut -d \| -f2 | cut -d \/ -f3)

File: pf9-express [hunk 3]
```diff
@@ -580,15 +580,15 @@ fi
 ## build group_vars/all.yml
 build_config --skip-inventory-check
 
-# validate all config values are set
```
Comment (mmccarre) [inline pf9-express]: This should remain in it's original spot. The modification you are looking for starts on new line 642 through 654 (possibly).

**Comment by mmccarre** [comment]
After a discussion with @jb4free, we are going to close this one and merge my new one since it was the branch that was most recently tested on.

**Pull request closed without merging**